base/regex.jl

```diff
@@ -178,9 +178,10 @@ function match(re::Regex, str::Union{SubString{String}, String}, idx::Integer, a
     end
     ovec = re.ovec
     n = div(length(ovec),2) - 1
-    mat = SubString(str, ovec[1]+1, ovec[2])
+    mat = SubString(str, ovec[1]+1, prevind(str, ovec[2]+1))
     cap = Union{Void,SubString{String}}[
-            ovec[2i+1] == PCRE.UNSET ? nothing : SubString(str, ovec[2i+1]+1, ovec[2i+2]) for i=1:n ]
+    ovec[2i+1] == PCRE.UNSET ? nothing : SubString(str, ovec[2i+1]+1,
```

Comment:
Indentation should be kept as this is inside an array comprehension.

Reply:
fixed